localstack-core/localstack/services/lambda_/event_source_mapping/pollers/sqs_poller.py

```diff
@@ -227,3 +250,45 @@ def message_attributes_to_lower(message_attrs):
         for key, value in dict(attr).items():
             attr[first_char_to_lower(key)] = attr.pop(key)
     return message_attrs
+
+
+def message_collector(
+    receive_fn: Callable[[...], dict], max_batch_size=10, max_batch_window=0.5
+) -> Generator[list, Any, None]:
+    """
+    Collects a batch of SQS messages, doing a ReceiveMessage call every iteration, allowing a returned batch to exceed 10 elements.
+
+    A batch is yielded when the size of the collection exceeds `max_batch_size` or the elapsed duration exceeds the `max_batch_window`.
+
+    :param receive_fn: A zero-arguments version of a `receieve_message` call.
+    :param max_batch_size: A batch is collected until this size limit is reached (corresponds to ESM's `BatchSize` parameter).
+    :param max_batch_window: A batch is collected until this duration has elapsed (corresponds to ESM's `MaximumBatchingWindowInSeconds` parameter).
+    :returns: A generator which returns a collected batch of messages if limits have been reached, else an empty-list, each iteration.
+    """
+    batch = []
+    start_t = time.monotonic()
+
+    while True:
+        time_elapsed = time.monotonic() - start_t
+        try:
+            response = receive_fn()
```

Comment:
Perhaps pass the number of messages we want to receive -- up to max size to prevent overflow.